utils/ixdtf/src/parsers/records.rs

```diff
@@ -0,0 +1,201 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! Parse records for `ixdtf`'s parsers to target.
+
+use alloc::vec::Vec;
+
+/// An `IsoParseRecord` is an intermediary record returned by ISO parsing functions.
+///
+/// `IsoParseRecord` is converted into the ISO AST Nodes.
+#[non_exhaustive]
+#[derive(Default, Debug, PartialEq)]
+pub struct IsoParseRecord<'a> {
+    /// Parsed Date Record
+    pub date: Option<DateRecord>,
+    /// Parsed Time
+    pub time: Option<TimeRecord>,
+    /// Parsed UtcOffset
+    pub offset: Option<UTCOffset>,
+    /// Parsed `TimeZone` data (UTCOffset | IANA name)
+    pub tz: Option<TimeZone<'a>>,
+    /// The parsed calendar value.
+    pub calendar: Option<&'a str>,
+    /// Annotations
+    pub annotations: Vec<Annotation<'a>>,
+}
+
+#[non_exhaustive]
+#[derive(Debug, Clone, PartialEq)]
+pub struct Annotation<'a> {
+    pub critical: bool,
+    pub key: &'a str,
+    pub value: &'a str,
+}
+
+#[non_exhaustive]
+#[derive(Default, Debug, Clone, Copy, PartialEq)]
+/// The record of a parsed date.
+pub struct DateRecord {
+    /// Date Year
+    pub year: i32,
+    /// Date Month
+    pub month: u8,
+    /// Date Day
+    pub day: u8,
+}
+
+/// Parsed Time info
+#[non_exhaustive]
+#[derive(Debug, Default, Clone, Copy, PartialEq)]
+pub struct TimeRecord {
+    /// An hour
+    pub hour: u8,
+    /// A minute value
+    pub minute: u8,
+    /// A second value.
+    pub second: u8,
+    /// A millisecond value
+    pub millisecond: u16,
+    /// A Microsecond value
+    pub microsecond: u16,
+    /// A Nanosecond value
+    pub nanosecond: u16,
```

Comment:
Suggestion (optional), here and below: I probably lean toward a single `nanosecond: u32` since (1) it is fewer bits overall, (2) there are fewer invariants to check, and (3) it is easier to reconstruct a `second` value.